src/Integration/ExceptionListenerIntegration.php

```diff
@@ -5,22 +5,38 @@
 namespace Sentry\Integration;
 
 use Sentry\ErrorHandler;
-use Sentry\State\Hub;
+use Sentry\State\HubAdapter;
+use Sentry\State\HubInterface;
 
 /**
  * This integration hooks into the global error handlers and emits events to
  * Sentry.
  */
 final class ExceptionListenerIntegration implements IntegrationInterface
 {
+    /**
+     * @var HubInterface The hub this integration is attached to
+     */
+    private $hub;
+
+    /**
+     * Constructor.
+     *
+     * @param HubInterface|null $hub The hub this integration is attached to
+     */
+    public function __construct(?HubInterface $hub = null)
+    {
+        $this->hub = $hub ?? HubAdapter::getInstance();
+    }
+
     /**
      * {@inheritdoc}
      */
     public function setupOnce(): void
     {
         $errorHandler = ErrorHandler::registerOnce(ErrorHandler::DEFAULT_RESERVED_MEMORY_SIZE, false);
-        $errorHandler->addExceptionHandlerListener(static function (\Throwable $exception): void {
-            Hub::getCurrent()->captureException($exception);
+        $errorHandler->addExceptionHandlerListener(function (\Throwable $exception): void {
+            $this->hub->captureException($exception);
```

Comment:
Here too, same behavioral change.

Reply:
This is a valid point and I just realized something else.
Apparently, no one reported this so far but:

We need to check if the integration is active on the client like we do in the ModulesIntegration for example.

https://github.com/getsentry/sentry-php/blob/62a48faf3fc4ae5b0564a0d87f7a91c47ea61ad9/src/Integration/RequestIntegration.php#L68-L72

The reason being is that if someone binds another client to the current hub, where this integration is not enabled we shouldn't report errors.

Reply:
So to clarify what we need to do:
```
    $hub = $this->hub  ?? HubAdapter::getInstance();
    $self = $hub->getIntegration(self::class); 
  
    if (!$self instanceof self) { 
        return $event; 
    } 

    $hub->captureException($exception);
```

Reply:
I reverted the change, however I will fix the bug in a separate PR targeting `2.1` so that we can release it earlier than the version this PR is targeting